### ISSUE TITLE
Support Windows with clink

### ISSUE THREAD

**Issue by eblis** [atuinsh/atuin#1811]
Is there any chance you might support Windows in a native format?

I'm not talking about mingw or WSL, but supporting Windows in batch (normal command prompt) with [clink](https://github.com/chrisant996/clink).

Clink brings a lot of features from the Linux world, it has history, history autocomplete, `CTLR + R` history search, optional integration with FZF (for history, folders, files, etc), lots of other things.

If any code changes are required in clink to support atuin the developer is very quick to respond.

**Comment by YummyOreo** [comment]
I dont think this is possible because cmd.exe does not have something like preexec and precmd that is needed for atuin. Please correct me if I'm wrong.

**Comment by eblis** [comment]
You can hook the [onhistory](https://chrisant996.github.io/clink/clink.html#clink.onhistory) callback so you can save the entry.
And I guess you'd hook the [onfiltermatches](https://chrisant996.github.io/clink/clink.html#clink.onfiltermatches) callback to provide the history suggestions on the fly.

Alternatively you could replace the `CTRL + R` hotkey to call your code and show the suggestions that way; I believe that's how [clink+fzf integration](https://github.com/chrisant996/clink-fzf?tab=readme-ov-file#key-bindings) is done - if you enable the fzf keybindings you get the fzf prompt when pressing `ctrl + r`.

**Comment by metajack** [comment]
Just FYI, if you use nushell on windows, atuin works just fine except that it puts config in a strange (for Windows) place of `~/.local/share/atuin`. I suspect there are appropriate hooks for powershell it could use as well.

However, to install it you need to `winget install protoc` and then get a new terminal so PATH changes take effect, then `cargo install atuin`. The `curl | sh` trick won't work, although I suspect it could be made to work if you take a look at how nushell does it.

**Comment by santagada** [comment]
nushell looks cool but cmd unfortunately is still the only shell that has access to all windows features and what bat files are written on so people can't easily switch. Is anyone working on this? I started using atuin on zsh and I'm hooked but my shell on windows is clink+cmd. If no one is doing it I can take a look as it shouldn't be too complicated.

**Comment by eblis** [comment]
I reverted to using click with fzf for history, more than enough for me at the moment. 
I find the fzf interface a bit better and I don't really have a need for sync so no loss for me

**Comment by spapas** [comment]
Hello friends, any news on this ? I think that, for starters, we can start by generating the atuin.exe binary for windows. If that binary exists then people can build from there and try to integrate with the clink. 

TIA

**Comment by Magniquick** [comment]
https://github.com/atuinsh/atuin/issues/135#issuecomment-2240945504 exists :)

**Comment by ltrzesniewski** [comment]
Ok, this hasn't been advertised a lot, but did you know Atuin supports [PowerShell on Windows](https://docs.atuin.sh/cli/guide/installation/#on-windows) now?

```powershell
winget install -e Atuinsh.Atuin
if (-not (Test-Path -Path $PROFILE)) { New-Item -ItemType File -Path $PROFILE -Force | Out-Null }
Write-Output 'atuin init powershell | Out-String | Invoke-Expression' >> $PROFILE
```

In any case, you have the native Atuin binary on WinGet or in the releases.

**Comment by markovejnovic** [comment]
We do not plan on supporting batch. [Microsoft appears to be blessing PS as the way forwards](https://www.theregister.com/software/2020/05/27/cmdexe-is-dead-long-live-powershell-microsoft-leads-aged-command-line-interpreter-out-into-maintenance-mode/461687). Working with `cmd.exe` is difficult, integrating atuin into it also seems technically challenging for very little pay-off in the future.

Thank you all!
